java/src/Ice/src/main/java/com/zeroc/IceInternal/Instance.java

```diff
@@ -761,10 +763,22 @@ public Class<?> apply(String typeId)
             if(pos != -1)
             {
                 String topLevelModule = typeId.substring(2, pos);
-                String pkg = _initData.properties.getProperty("Ice.Package." + topLevelModule);
-                if(pkg.length() > 0)
+                String[] packages = _builtInModulePackages.get(topLevelModule);
+                if (packages == null)
                 {
-                    c = getConcreteClass(pkg + "." + className);
+                    packages = _initData.properties.getPropertyAsListWithDefault("Ice.Package." + topLevelModule, null);
+                }
+
+                if(packages != null)
+                {
+                    for(String pkg : packages)
+                    {
+                        c = getConcreteClass(pkg + "." + className);
```

Comment:
I am confused here. 

My understanding is `pkg` represents a prefix, such as  `com.zeroc`. Does it mean className includes part of the actual package name, like `Ice.AdapterNotFoundException`?

Reply:
className is the fully qualified name, without package prefix derived from java:package metadata.